.github/workflows/ci-wheels.yml

```diff
@@ -0,0 +1,102 @@
+# A workflow to build, test and publish PyPI sdist/bdist artifacts
+#
+# Reference:
+#   - https://github.com/actions/checkout
+#   - https://github.com/actions/download-artifact
+#   - https://github.com/actions/setup-python
+#   - https://github.com/actions/upload-artifact
+#   - https://github.com/pypa/build
+#   - https://github.com/pypa/gh-action-pypi-publish
+#   - https://test.pypi.org/help/#apitoken
+
+name: ci-wheels
+
+on: [pull_request, push]
+
+
+jobs:
+  build-artifacts:
+    name: "Build PyPI artifacts"
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+      with:
+        fetch-depth: 0
+
+    - name: "Build sdist/bdist"
+      run: |
+        # ugrid-checks is a pure python package, so simply use pypa/build
+        pipx run build
+
+    - name: "Show sdist/bdist"
+      run: |
+        ls -l ${{ github.workspace }}/dist
+
+    - uses: actions/upload-artifact@v2
+      with:
+        name: pypi-artifacts
+        path: ${{ github.workspace }}/dist
+
+
+  test-artifacts:
+    needs: [build-artifacts]
+    name: "Test bdist"
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v2
+
+    - uses: actions/setup-python@v2
+      with:
+        python-version: "3.10"
```

Comment:
@pp-mo As this is a pure Python bdist, it's sufficient to only build and test on one version (the latest stable) of Python.